sos/plugins/__init__.py

```diff
@@ -180,6 +180,40 @@ def is_installed(self, package_name):
         '''Is the package $package_name installed?'''
         return self.policy().pkg_by_name(package_name) is not None
 
+    def do_cmd_private_sub(self, cmd):
+        '''Remove certificate and key output archived by sosreport. cmd
+        is the command name from which output is collected (i.e. exlcuding
+        parameters). Any matching instances are replaced with: '-----SCRUBBED'
+        and this function does not take a regexp or substituting string.
+
+        This function returns the number of replacements made.
+        '''
+        globstr = '*' + cmd + '*'
+        self._log_debug("Scrubbing certs and keys for commands matching %s"
+                        % (cmd))
+
+        if not self.executed_commands:
+            return 0
+
+        replacements = None
+        try:
+            for called in self.executed_commands:
+                if called['file'] is None:
+                    continue
+                if fnmatch.fnmatch(called['exe'], globstr):
+                    path = os.path.join(self.commons['cmddir'], called['file'])
+                    readable = self.archive.open_file(path)
```

Comment:
I think we will be revisiting this API before 3.4 anyway - it's not exactly what I had in mind when we were discussing it (I only really merged it today as we are under such pressure of time and since it solves a real obfuscation problem for that plugin).

What this *should* be doing is providing a /generic/ block find/replace feature - exactly what we discussed when this first came up.

This is a slightly weird/specialized interface that only addresses one possible case of many - a general interface needs to accept the start and end block markers as arguments rather than hard-coding them in this manner.